### PR TITLE
Add API

### PR DESCRIPTION
**Pull request by HandlessMidas**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+func (a *Api) Router() http.Handler {
+	router := mux.newRouter
+
+	router.HandleFunc("/register", a.register).Methods(http.MethodPost)
+	router.HandleFunc("/login", a.login).Methods(http.MethodPost)
+	router.HandleFunc("/logout", a.logout).Methods(http.MethodPost)
+
+	router.HandleFunc("/docs/{doc_id}/get", a.getDoc).Methods(http.MethodGet)
```
Comment (M0nsteR) [inline api/api.go]: Please use the RESTful approach here, mapping your routes to HTTP verbs. No need to use "/delete" and similar suffixes.

File: model/model.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package model
+
+type UseCasesInterface interface {
+	Register(request LoginRequest) (*LoginResponse, error)
+	Login(request LoginRequest) (*LoginResponse, error)
+	Logout(token Token) error
+
+	CreateDoc(token Token, doc Doc) (*Doc, error)
```
Comment (M0nsteR) [inline model/model.go]: Tokens should be handled by middleware. All usecase functions should operate with some kind of user account, 
specifying user id, roles and access claims, but not JWT!

**Review by M0nsteR — CHANGES_REQUESTED** [review]

File: api/api.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	router.HandleFunc("/register", a.register).Methods(http.MethodPost)
+	router.HandleFunc("/login", a.login).Methods(http.MethodPost)
+	router.HandleFunc("/logout", a.logout).Methods(http.MethodPost)
+
+	router.HandleFunc("/docs/{doc_id}", a.getDoc).Methods(http.MethodGet)
+	router.HandleFunc("/docs/{doc_id}", a.createDoc).Methods(http.MethodPost)
+	router.HandleFunc("/docs/{doc_id}", a.deleteDoc).Methods(http.MethodDelete)
+	router.HandleFunc("/docs/{doc_id}", a.editDoc).Methods(http.MethodPut)
+
+	router.HandleFunc("/docs/get_all_docs", a.getAllDocs).Methods(http.MethodGet)
```
Comment (NekoNe) [inline api/api.go]: May be just GET: /docs ?

File: api/api.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+	router.HandleFunc("/users", a.editUser).Methods(http.MethodPut)
+	router.HandleFunc("/users", a.getUser).Methods(http.MethodGet)
+
+	router.HandleFunc("/users/friends", a.getFriends).Methods(http.MethodGet)
```
Comment (NekoNe) [inline api/api.go:35]: Is this method for retrieving all friends for a specific user?

Comment (ilyavoronin) [reply api/api.go:35]: No, user's friends aren't visible for other users

File: api/api.go (hunk 3 above)
Comment (ilyavoronin) [reply api/api.go]: fixed

**Review by M0nsteR — APPROVED** [review]
Looks better, but you still need to refactor your code to catch up with the latest chat example provided by Nikita.

**Pull request merged**